pysr/regressor_sequence.py

```diff
@@ -0,0 +1,192 @@
+from typing import List, Optional, Union
+
+import numpy as np
+
+from .sr import PySRRegressor
+from .utils import ArrayLike
+
+
+def _check_assertions(
+    X,
+    y=None,
+    Xresampled=None,
+    recursive_history_length=None,
+    weights=None,
+    variable_names=None,
+    X_units=None,
+    y_units=None,
+):
+    if recursive_history_length is not None and recursive_history_length <= 0:
+        raise ValueError(
+            "The `recursive_history_length` parameter must be greater than 0 (otherwise it's not recursion)."
+        )
+    if len(X.shape) > 2:
+        raise ValueError(
+            "Recursive symbolic regression only supports up to 2D data; please flatten your data first"
+        )
+    elif len(X) < 2:
+        raise ValueError(
+            "Recursive symbolic regression requires at least 2 datapoints; if you tried to pass a 1D array, use array.reshape(-1, 1)"
+        )
+    if len(X) <= recursive_history_length + 1:
+        raise ValueError(
+            f"Recursive symbolic regression with a history length of {recursive_history_length} requires at least {recursive_history_length + 2} datapoints."
+        )
+    if isinstance(weights, np.ndarray) and len(weights) != len(X):
+        raise ValueError("The length of `weights` must have shape (n_times,).")
+    if isinstance(variable_names, list) and len(variable_names) != X.shape[1]:
+        raise ValueError(
+            "The length of `variable_names` must be equal to the number of features in `X`."
+        )
+    if isinstance(X_units, list) and len(X_units) != X.shape[1]:
+        raise ValueError(
+            "The length of `X_units` must be equal to the number of features in `X`."
+        )
+    if y is not None:
+        raise ValueError("Recursive symbolic regression does not use `y`")
+    if y_units is not None:
+        raise ValueError("Recursive symbolic regression does not use `y_units`")
+    if Xresampled is not None:
+        raise ValueError("Recursive symbolic regression does not use `Xresampled`")
+
+
+class PySRSequenceRegressor(PySRRegressor):
+    def __init__(
+        self,
+        recursive_history_length: int = 0,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.recursive_history_length = recursive_history_length
+
+    def _variable_names(self, y, variable_names=None):
+        if not variable_names:
+            if y.shape[1] == 1:
```

Comment:
Since only `y.shape[1]` is used, it would be easier and more readable to pass `n_features: int` here rather than all of `y` as an argument.